File: src/Entity/Operation.php
<?php

namespace App\Entity;

use App\Repository\OperationRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: OperationRepository::class)]
class Operation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    private ?int $id_user = null;

    #[ORM\Column(length: 100)]
    #[Assert\NotBlank(message: 'Le type d\'opération est obligatoire.')]
    #[Assert\Length(min: 5, minMessage: 'Le type doit contenir au moins {{ limit }} caractères.')]
    #[Assert\Regex(
        pattern: '/^[\p{L}0-9 ]+$/u',
        message: 'Le type ne doit pas contenir de caractères spéciaux.'
    )]
    private ?string $type_operation = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotNull(message: 'La date de début est requise.')]
    #[Assert\GreaterThanOrEqual(
        value: 'today',
        message: 'La date de début ne peut pas être dans le passé.'
    )]
    private ?\DateTime $date_debut = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotNull(message: 'La date de fin est requise.')]
    #[Assert\GreaterThanOrEqual(
        propertyPath: 'date_debut',
        message: 'La date de fin ne peut pas être avant la date de début.'
    )]
    private ?\DateTime $date_fin = null;

    #[ORM\Column(length: 50)]
    private ?string $statut = null;

    #[ORM\ManyToOne(inversedBy: 'operations')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Equipement $equipement = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getIdUser(): ?int
    {
        return $this->id_user;
    }

    public function setIdUser(int $id_user): static
    {
        $this->id_user = $id_user;

        return $this;
    }

    public function getTypeOperation(): ?string
    {
        return $this->type_operation;
    }

    public function setTypeOperation(string $type_operation): static
    {
        $this->type_operation = $type_operation;

        return $this;
    }

    public function getDateDebut(): ?\DateTimeInterface
{
    return $this->date_debut;
}

public function setDateDebut(?\DateTimeInterface $date_debut): static
{
    $this->date_debut = $date_debut;
    return $this;
}

public function getDateFin(): ?\DateTimeInterface
{
    return $this->date_fin;
}

public function setDateFin(?\DateTimeInterface $date_fin): static
{
    $this->date_fin = $date_fin;
    return $this;
}

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): static
    {
        $this->statut = $statut;

        return $this;
    }

    public function getEquipement(): ?Equipement
    {
        return $this->equipement;
    }

    public function setEquipement(?Equipement $equipement): static
    {
        $this->equipement = $equipement;

        return $this;
    }
}

File: src/main/java/entities/Operation.java
package entities;

import java.sql.Date;

public class Operation {
    private int id;
    private int equipement_id;
    private int id_user;
    private String nomEquipement;
    private String TypeOperation;
    private Date date_debut ;
    private Date date_fin ;
    private String statut;


    public Operation(int id_operation, int equipement_id, int id_user, String type_operation, Date date_debut, Date date_fin, String statut) {
        this.id = id_operation;
        this.equipement_id = equipement_id;
        this.id_user = id_user;
        this.TypeOperation = type_operation;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.statut = statut;

    }
    public Operation(String type_operation, Date date_debut, Date date_fin, String statut) {
        this.equipement_id = equipement_id;
        this.TypeOperation = type_operation;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.statut = statut;

    }

    public Operation(int id_user, String type_operation, Date date_debut, Date date_fin, String statut) {
        this.id_user = id_user;
        this.TypeOperation = type_operation;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.statut = statut;
    }

    public Operation() {

    }

    public int getId_operation() {
        return id;
    }
    public void setId_operation(int id_operation) {
        this.id = id_operation;
    }
    public String getType_operation() {
        return TypeOperation;
    }
    public void setType_operation(String type_operation) {
        this.TypeOperation = type_operation;
    }
    public Date getDate_debut() {
        return date_debut;
    }
    public void setDate_operation(Date date_operation) {
        this.date_debut = date_operation;
    }
    public String getStatut() {
        return statut;
    }
    public void setStatut(String statut) {
        this.statut = statut;
    }
    public void setDate_debut(Date date_debut) {
        this.date_debut = date_debut;
    }
    public Date getDate_fin() {
        return date_fin;
    }
    public void setDate_fin(Date date_fin) {
        this.date_fin = date_fin;
    }
    public int getEquipement_id() {
        return equipement_id;
    }
    public void setEquipement_id(int equipement_id) {
        this.equipement_id = equipement_id;
    }
    public String getNomEquipement() {
        return nomEquipement;
    }
    public void setNomEquipement(String nomEquipement) {
        this.nomEquipement = nomEquipement;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    @Override
    public String toString() {
        return "Operation : " +
                " id_equipement = '" + equipement_id + '\'' +
                " id_user = '" + id_user + '\'' +
                " type_operation = '" + TypeOperation + '\'' +
                " Date Debut = " + date_debut +
                " Date Fin = " + date_fin +
                " statut = '" + statut + "'\n"
                ;
    }

}
